pylint/checkers/classes.py

```diff
@@ -133,6 +133,39 @@ def _positional_parameters(method):
         positional = positional[1:]
     return positional
 
+def _has_different_parameters_default_value(original, overridden):
+    """
+    Check if original and overridden methods arguments have the same default values
+
+    Return True if one of the overridden arguments has a default
+    value different from the default value of the original argument
+    If one of the method doesn't have argument (.args is None)
+    return False
+    """
+    if original.args is None or overridden.args is None:
+        return False
+
+    original_param_names = [param.name for param in original.args]
+    default_missing = object()
+    for param_name in original_param_names:
+        try:
+            original_default = original.default_value(param_name).value
+        except AttributeError:
+            # if value attribute doesn't exist, it may be due to the fact
+            # that the default value is a ClassDef object and then the default
+            # value is taken to be the string representation
+            original_default = original.default_value(param_name).as_string()
```

Comment:
I would say we should refactor this a bit to avoid some potential expensive calls:

1. just get the default value, without accessing ``.value`` for both of the fields. thus when getting to line 165, we had no ``as_string`` call thus far
2. there, check for common node kinds, by verifying their type. E.g, are these two Consts? get their value and check against that. Are these classes, get their name and check against that. We might throw the class type in there as well, so we can differentiate between List and Tuples for example. Calling ``as_string`` might be potentially slow which is why I prefer to avoid doing it unless necessary